### PR TITLE
docs(LOG): Added documentation

### LINKED ISSUE #1653: Add Log error levels to documentation & document "setLevel"
**Browser**: Chrome (latest)

**OS**: Windows 10

**Clappr Version**: 0.2.95

Going through all issues related to logging, it seems like someone insisted on having fatal errors logged without an open to turn them off.

I'd appreciate providing of any way to disable also fatal errors, as it makes no sense to have them on production.

Also, the documentation (API reference) does not supply any info on "Clappr.Log"

### PR DESCRIPTION
**Pull request by MarcosEllys**
Added Log Plugin documentation

closes #1653

### CODE REVIEW

File: src/plugins/log/log.js [hunk 1]
```diff
@@ -19,7 +19,34 @@ const LEVEL_DISABLED = LEVEL_ERROR
...
+ * Clappr.Log.setLevel(0);
+ * // Info Log
+ * Clappr.Log.setLevel(1);
+ * // Warn Log
+ * Clappr.Log.setLevel(2);
+ * // Error Log
+ * Clappr.Log.setLevel(3);
+ * // Debug Log
+ * Clappr.Log.setLevel(4);
+ * ```
```
Comment (towerz) [inline src/plugins/log/log.js]: Please don't use the log levels as their numeric values. Instead, use their constants, which are defined as: `LEVEL_DEBUG`, `LEVEL_INFO`, `LEVEL_WARN` and `LEVEL_ERROR`. For some reason, the `LEVEL_DISABLED` wasn't exposed, so please add it as well.

File: src/plugins/log/log.js [hunk 2]
```diff
@@ -19,7 +19,34 @@ const LEVEL_DISABLED = LEVEL_ERROR
...
+ */
 export default class Log {
+
+  /**
+   * @method constructor
+   * @param {Number} level should be an integer with the  value 1, 2, 3, 4.
+   * @param {Number} offLevel should be an integer with the  value 1, 2, 3, 4.
+   */
```
Comment (towerz) [inline src/plugins/log/log.js]: As before, don't use the numeric values, but the constants instead.

File: src/plugins/log/log.js [hunk 3]
```diff
@@ -81,9 +113,30 @@ Log.getInstance = function() {
   return this._instance
 }
 
+/**
+ * Specify a `level` to log.
+ * @method setLevel
+ * @param {Number} level valid level log [0,1,2,3,4].
+ */
```
Comment (towerz) [inline src/plugins/log/log.js]: I'd describe this as "configure the minimum level to be logged". The `setLevel` only restricts the logs up to a certain value, logging whatever message with a higher log level. So, when set to `LEVEL_INFO` the log would display messages from `LEVEL_INFO` up to `LEVEL_ERROR`.

File: src/plugins/log/log.js [hunk 4]
```diff
@@ -81,9 +113,30 @@ Log.getInstance = function() {
...
 Log.info = function() { this.getInstance().info.apply(this.getInstance(), arguments) }
+/**
+ * Set Log for Warn.
+ * @method warn
+ */
 Log.warn = function() { this.getInstance().warn.apply(this.getInstance(), arguments) }
+/**
+ * Set Log for Error.
+ * @method error
+ */
```
Comment (towerz) [inline src/plugins/log/log.js]: Those methods are only static helpers to log messages with a given log level, not "setting the log" to a given level.

**Comment by towerz** [comment]
Since this PR wasn’t updated, I’ll close it for now. Please open a new one or reopen if necessary.

**Comment by asheroto** [comment]
Hello,

Can you please clarify how to set the log level to LEVEL_DISABLED, LEVEL_INFO, etc?  I can't seem to find those vars.

Can you actually disable the logging or has that not been implemented yet?

Thanks

**Pull request closed without merging**